scripts/dotnet-cli-build/PrepareTargets.cs

```diff
@@ -200,6 +181,172 @@ public static BuildTargetResult RestorePackages(BuildTargetContext c)
             return c.Success();
         }
 
+        [Target]
+        [BuildPlatforms(BuildPlatform.Ubuntu)]
+        public static BuildTargetResult CheckUbuntuDebianPackageBuildDependencies(BuildTargetContext c)
+        {
+            var debianPackageBuildDependencies = new string[]
+            {
+                "devscripts", 
+                "debhelper",
+                "build-essential"
+            };
+
+            var messageBuilder = new StringBuilder();
+
+            foreach (var package in debianPackageBuildDependencies)
+            {
+                if (!AptPackageIsInstalled(package))
+                {
+                    messageBuilder.Append($"Error: Debian package build dependency {package} missing.");
+                    messageBuilder.Append(Environment.NewLine);
+                    messageBuilder.Append($"-> install with apt-get install {package}");
+                    messageBuilder.Append(Environment.NewLine);
+                }
+            }
+
+            if (messageBuilder.Length == 0)
+            {
+                return c.Success();
+            }
+            else
+            {
+                return c.Failed(messageBuilder.ToString());
+            }
+        }
+
+        [Target]
+        [BuildPlatforms(BuildPlatform.Ubuntu)]
+        public static BuildTargetResult CheckUbuntuCoreclrDependencies(BuildTargetContext c)
+        {
+            var errorMessageBuilder = new StringBuilder();
+
+            var ubuntuCoreclrDependencies = new string[]
+            {
+                "unzip",
+                "curl",
+                "libicu-dev",
+                "libunwind8",
+                "gettext",
+                "libssl-dev",
+                "libcurl3-gnutls",
+                "zlib1g",
+                "liblttng-ust-dev",
+                "lldb-3.6-dev",
```

Comment:
As Matt mentioned, are these intended to be dependencies to run coreclr, to compile coreclr, or compile the entire coreclr repo? Unless its the last one I doubt you want lldb-3.6 or lldb-3.6-dev. These packages are a debugger and debugger plugin sdk respectively. They are used so that we can compile libsosplugin.so, our .Net extension to lldb that helps with debugging.